fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -105,9 +109,12 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 			double getTimeEstimate() override { return SERVER_KNOBS->COMMIT_TIME_ESTIMATE; }
 		};
 		void action(CloseAction& a) {
-			auto s = db->Close();
-			if (!s.ok()) {
-				TraceEvent(SevError, "RocksDBError").detail("Error", s.ToString()).detail("Method", "Close");
+			if (store->db) {
+				auto s = store->db->Close();
+				if (!s.ok()) {
+					TraceEvent(SevError, "RocksDBError").detail("Error", s.ToString()).detail("Method", "Close");
+				}
+				store->db = nullptr;
```

Comment:
I _think_ this will cause the memory associated with the DB to get leaked. Right now, we free the memory for the DB in `~Writer()`, but that won't run with this change.

Reply:
No; you're right.  Will fix.